pkg/component/component.go

```diff
@@ -310,3 +332,14 @@ func (c *Component) Close() {
 func (c *Component) AllowInsecureForCredentials() bool {
 	return c.config.GRPC.AllowInsecureForCredentials
 }
+
+// ServeHTTP serves an HTTP request.
+// If the Content-Type is application/grpc, the request is routed to the gRPC gateway.
+// Otherwise, the request is routed to the default web server.
+func (c *Component) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
https://github.com/TheThingsNetwork/lorawan-stack/pull/204#issue-256669105
> **Notes for Reviewers:**
>
> Confirm that this is the right direction.
>
> Note that this a draft PR; no need for small code style things if any.

Reply:
Because this is not specific to web anymore; it's a generic HTTP request that goes in web (for OAuth, Console, static, etc) or gRPC (equal level systems).